provider_shopper/test/cart_widget_test.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:provider/provider.dart';
+import 'package:provider_shopper/models/cart.dart';
+import 'package:provider_shopper/models/catalog.dart';
+import 'package:provider_shopper/screens/cart.dart';
+
+CartModel cartModel;
+CatalogModel catalogModel;
+Widget createCartScreen() => MultiProvider(
+      providers: [
+        Provider(create: (context) => CatalogModel()),
+        ChangeNotifierProxyProvider<CatalogModel, CartModel>(
+          create: (context) => CartModel(),
+          update: (context, catalog, cart) {
+            catalogModel = catalog;
+            cartModel = cart;
+            cart.catalog = catalogModel;
+            return cart;
+          },
+        ),
+      ],
+      child: MaterialApp(
+        home: MyCart(),
+      ),
+    );
+
+void main() {
+  group('CartScreen widget tests', () {
+    testWidgets('Testing when the cart is empty', (tester) async {
+      await tester.pumpWidget(createCartScreen());
+
+      // Testing the layout of the screen widgets
+      expect(find.text('Cart'), findsOneWidget);
```

Comment:
The strength of this test is in lines 45-50, which are a great check that the behavior of the BUY button works as designed.

In order to keep the focus of this test as narrow as possible, I would remove these three expect statements and change the name of the test to "Tapping BUY button displays snackbar."